File: 2020/day3/day3.exs
lines =
  File.read!("input.txt")
  |> String.split("\n")

trees =
  Enum.with_index(lines)
  |> Enum.reduce([], fn {line, y}, acc ->
    String.graphemes(line)
    |> Enum.with_index()
    |> Enum.reduce(acc, fn {char, x}, acc ->
      case char do
        "." -> acc
        "#" -> [{x, y} | acc]
      end
    end)
  end)

is_tree = &Enum.any?(trees, fn tree -> tree == &1 end)

slope = fn {r, d} ->
  Enum.count(
    0..Enum.count(lines),
    &is_tree.({rem(&1 * r, Enum.at(lines, 0) |> String.length()), &1 * d})
  )
end

Enum.product_by([{1, 1}, {3, 1}, {5, 1}, {7, 1}, {1, 2}], slope) |> IO.inspect()

File: 2016/day3/day3.py
with open("input.txt") as f:
    lines = f.readlines()

total = 0

triangles = []
t1, t2, t3 = [], [], []

for line in lines:
    a, b, c = map(int, line.split())

    t1.append(a)
    t2.append(b)
    t3.append(c)

    if len(t1) >= 3:
        triangles.append(t1)
        triangles.append(t2)
        triangles.append(t3)
        t1 = []
        t2 = []
        t3 = []

for triangle in triangles:
    a, b, c = triangle
    if a + b > c and a + c > b and b + c > a:
        total += 1

print(total)
